akka-actor/src/main/scala/akka/actor/Address.scala

```diff
@@ -75,7 +75,11 @@ object Address {
   /**
    * Constructs a new Address with the specified protocol, system name, host and port
    */
-  def apply(protocol: String, system: String, host: String, port: Int) = new Address(protocol, system, Some(host), Some(port))
+  def apply(protocol: String, system: String, host: String, port: Int) = {
+    assert(!host.contains(":") || host.startsWith("["), s"IPv6 address $host must be wrapped with [] to be safe")
```

Comment:
assert is **deadly** :fire: Please use `require` instead.

// `assert` throws `AssertionError`, which is an `Error`, not exception, which can cause the entire `ActorSystem` to blow up, because `Error` is not `NonFatal`